File: Practices/Strings/LCM.java
import java.util.Scanner;

public class LCM {
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        String lcmString;

        System.out.println("Enter number of test cases you want to run: ");
        int numberOfTestcases = scan.nextInt();
        scan.nextLine();

        for(int i=0;i<numberOfTestcases;i++){
            System.out.println("Enter Frist String: ");
            String str1 = scan.nextLine();
            System.out.println("Enter Second String: ");
            String str2 = scan.nextLine();
            lcmString = stringLcm(str1,str2);
            System.out.println(lcmString);
        }

        scan.close();
        
    }

    static int lcm(int num1, int num2){
        return (num1*num2)/(HCF(num1, num2));       
    }

    static int HCF(int num1,int num2){
        if( num2==0 )
            return num1;
        else
            return HCF(num2, num1%num2);
    }

    static String stringLcm(String str1, String str2){
        int str1_length = str1.length();
        int str2_length = str2.length();
        int lengthOfLcmString = lcm(str1_length,str2_length);
        String lcmString = "";


        for(int i=0;i<lengthOfLcmString;i++){
            if(str1.charAt(i%str1_length) != str2.charAt(i%str2_length)){
                return "-1";
            }
        }

        for(int i=0 ; i< lengthOfLcmString/str1_length; i++){
            lcmString += str1;
        }

        return lcmString;
    }
}

File: Practices/Strings/LCM.cpp
// codeforces problem "B. String LCM"
#include<iostream>
#include<string>
using namespace std;

int lcm(int a,int b);
int hcf(int a,int b);
string* stringLcm(string& str1,string& str2);

int main(){
    string S,T;
    string* lcm_str;

    cout<<"Enter first string: ";
    getline(cin,S);
    cout<<"Enter second string: ";
    getline(cin,T);

    lcm_str = stringLcm(S,T);
    cout<<*lcm_str;
}

int hcf(int a,int b){
    if(b==0)
        return a;
    else
        return hcf(b,a%b);
}

int lcm(int a, int b){
    return (a*b)/hcf(a,b);
}

string* stringLcm(string& str1,string& str2){
    int n = str1.size();
    int m = str2.size();
    int lng_of_lcm_str = lcm(n,m);
    string* lcm_str = new string;
    for(int i=0;i<lng_of_lcm_str/n;i++)
        *lcm_str += str1;
    
    for(int i=0; i<lng_of_lcm_str;i++){
        if( str1[i%n] != str2[i%m]){
            *lcm_str = "-1";
            break;
        }
    }
    return lcm_str;
}